pkg/sdk/roles_integration_test.go

```diff
@@ -0,0 +1,323 @@
+package sdk
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInt_Roles(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	database, databaseCleanup := createDatabase(t, client)
+	t.Cleanup(databaseCleanup)
+	schema, _ := createSchema(t, client, database)
+	tag, _ := createTag(t, client, database, schema)
+	tag2, _ := createTag(t, client, database, schema)
+
+	t.Run("create no options", func(t *testing.T) {
+		roleID := randomAccountObjectIdentifier(t)
+		err := client.Roles.Create(ctx, roleID, nil)
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			err := client.Roles.Drop(ctx, roleID, nil)
+			require.NoError(t, err)
+		})
+
+		role, err := client.Roles.ShowByID(ctx, roleID)
+		require.NoError(t, err)
+
+		assert.Equal(t, roleID.Name(), role.Name)
+	})
+
+	t.Run("create if not exists", func(t *testing.T) {
+		roleID := randomAccountObjectIdentifier(t)
+		err := client.Roles.Create(ctx, roleID, &CreateRoleOptions{
+			IfNotExists: Bool(true),
+		})
+		require.NoError(t, err)
+		t.Cleanup(func() {
```

Comment:
This cleanup is repeated, and it is used also in helper_test.go, so maybe extract it and reuse it?

Reply:
I actually created createRoleWithOptions which creates role with passed options and it returns the cleanup function. I just didn't see that it's not used everywhere

Reply:
my question is -- do we even need separate functions for createXXX vs createXXXWithOptions? We almost never us the createXXXWithOptions. How about we just have createXXX

Reply:
I used it a few times in this test and I think we can leave with two create functions since creating createXXX function is as simple as calling the one with opts but passing nil or default options there. We can do either, but like I said I used those two in roles_integration_test.go go get the cleanup function for free, so I didn't need to copy cleanup functions from other tests